crates/module-system/sov-state/src/nomt/prover_storage.rs

```diff
@@ -599,11 +601,13 @@ where
         };
         let storage_root_historical = self.get_root_hash_unbound(version_to_use)?;
         if self.should_check_dbs_sync(version_to_use) {
-            let user_session = self.state_session_builder.begin_user_session()?;
+            let SessionsContainer {
+                user: user_session,
+                kernel: kernel_session,
+            } = self.state_session_builder.begin_both_sessions()?;
             let user_root = user_session.prev_root();
-            drop(user_session);
-            let kernel_session = self.state_session_builder.begin_kernel_session()?;
             let kernel_root = kernel_session.prev_root();
+            drop(user_session);
             drop(kernel_session);
```

Comment:
What if we don’t desugar user_session and simply call `drop(user_session)` instead of dropping the two sessions separately?